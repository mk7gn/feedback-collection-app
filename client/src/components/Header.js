import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

class Header extends Component {
  renderContent() {
    const {auth} = this.props;
    if (auth === null) {
      return;
    }
    if (auth === false) {
      return (
        <ul id="nav-mobile" className="right hide-on-med-and-down">
          <li><a href="">Login</a></li>
          <li><a href="/auth/google">Login with Google</a></li>
          <li><a href="/auth/facebook">Login with Facebook</a></li>
        </ul>
      );
    }
    return (
      <ul id="nav-mobile" className="right hide-on-med-and-down">
        <li>{auth.firstName} {auth.lastName}</li>
        <li><a href="/auth/logout">Logout</a></li>
      </ul>
    );
  }

  render() {
    return (
      <nav className="teal lighten-2">
        <div className="nav-wrapper row">
          <Link to={this.props.auth ? '/surveys' : '/'}
                className="brand-logo col m6">
            <i className="material-icons">email</i>Emaily
          </Link>
          <div className="col m6 right">
            { this.renderContent() }
          </div>
        </div>
      </nav>
    );
  }
}

function mapStateToProps({auth}) {
  return { auth };
};

export default connect(mapStateToProps)(Header);
