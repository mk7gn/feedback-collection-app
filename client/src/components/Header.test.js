import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Header from './Header';

afterEach(cleanup);

test('should render component', () => {
  expect(true).toBeTruthy();
});