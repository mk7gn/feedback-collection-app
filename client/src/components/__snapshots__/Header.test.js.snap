// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="nav-wrapper"
    >
      <a
        class="brand-logo"
        href="/"
      >
        Emaily
      </a>
      <ul
        class="right hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            href="/auth/google"
          >
            Login with Google
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
